File: app/src/main/java/com/weathercock/cleanarchitecture/domain/repository/WeatherRepository.java
package com.weathercock.cleanarchitecture.domain.repository;

import com.weathercock.cleanarchitecture.domain.model.weather.WeatherModel;

import rx.Observable;

/**
 * Weather Repository
 *
 * @author Kensuke
 */
public interface WeatherRepository {

    Observable<WeatherModel> fetch(String cityId);

    WeatherModel find(String cityId);

}

File: domain/src/main/java/com/kazakago/cleanarchitecture/domain/repository/weather/WeatherRepository.kt
package com.kazakago.cleanarchitecture.domain.repository.weather

import com.kazakago.cleanarchitecture.domain.model.city.CityId
import com.kazakago.cleanarchitecture.domain.model.weather.Weather

interface WeatherRepository {

    suspend fun find(cityId: CityId): Weather

    suspend fun insert(weather: Weather)

    suspend fun delete(weather: Weather)

}
